hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerController.java

```diff
@@ -79,9 +84,18 @@ public String getContainerLocation(final long containerId) {
   public void markContainerForClose(final long containerId)
       throws IOException {
     Container container = containerSet.getContainer(containerId);
-
-    if (container.getContainerState() == State.OPEN) {
-      getHandler(container).markContainerForClose(container);
+    if (container==null) {
+      Set<Long> missingContainerSet = containerSet.getMissingContainerSet();
+      if (missingContainerSet.contains(containerId)) {
+        LOG.warn("The Container is in the MissingContainerSet " +
+                "hence we can't close it. ContainerID: {}", containerId);
+      } else {
+        LOG.warn("The Container is not found. ContainerID: {}", containerId);
+      }
+    } else {
+      if (container.getContainerState() == State.OPEN) {
+        getHandler(container).markContainerForClose(container);
+      }
```

Comment:
This is an usually case that container is not found.  I would suggest throw an container not found storage exception here,  the outer has the logic to handle the exception.

Reply:
Thank you for your input. I have place Log here so that we could have a better understanding of whether the container is present in the MissingContainerSet or not which allows us to debug better. @avijayanhwx your thoughts on that?

Reply:
+1 to @ChenSammi's comment. We should throw a Storage Container Exception here, that can be handled by callers (notifyGroupRemove & CloseContainerCommandHandler)